gm4_balloon_animals/data/gm4_balloon_animals/functions/wandering_trader/check_eligibility.mcfunction

```diff
@@ -0,0 +1,22 @@
+# Checks the eligibility of the wandering trader to become an animal trader
+# @s = a wandering trader
+# at @s
+# run from #gm4_trades:register_trades
+
+scoreboard players set $valid_trader gm4_balloon_animals_data 1
+
+# check tags
+execute if entity @s[tag=gm4_balloon_animal_eligible_ignore] run scoreboard players set $valid_trader gm4_balloon_animals_data 0
+execute if entity @s[tag=gm4_calling_bell_trader] run scoreboard players set $valid_trader gm4_balloon_animals_data 0
+
+data modify storage gm4_balloon_animals:temp trader.uuid set from entity @s UUID
+tag @s add gm4_balloon_animal_trader_eligible_check
+
+# must have 2 llamas leashed
+scoreboard players set $trader_llama_count gm4_balloon_animals_data 0
+execute as @e[type=trader_llama,distance=..6] on leasher if entity @s[tag=gm4_balloon_animal_trader_eligible_check] run scoreboard players add $trader_llama_count gm4_balloon_animals_data 1
```

Comment:
You may want to include a smithed tag here to validate that the trader llamas are actually normal trader llamas. However, we'll have to do some digging to see which one that would be.